generate-spec.php

```diff
@@ -757,6 +757,36 @@
 		if ($route->isPublic) {
 			// Add empty authentication, meaning that it's optional. We can't know if there is a difference in behaviour for authenticated vs. unauthenticated access on public pages (e.g. capabilities)
 			$security[] = new stdClass();
+		} else {
+			$content = [
+				'application/json' => [
+					'schema' => $route->isOCS
+						? Helpers::wrapOCSResponse(new stdClass())
+						: [
+							'type' => 'object',
+							'required' => [
+								'message',
+							],
+							'properties' => [
+								'message' => [
+									'type' => 'string',
+								],
+							],
+						],
+				],
+			];
+
+			$mergedResponses[401] ??= [
+				'description' => 'Current user is not logged in',
+				'content' => $content,
+			];
+
+			if ($route->isAdmin) {
+				$mergedResponses[403] ??= [
```

Comment:
This needs more work, as the endpoint might already define 401/403 status codes, but with different return types. Therefore it needs to be done before the responses merging logic above.